File: 01/part2.py
input = open('input.txt').readlines()

increases = 0

for i in range(len(input)-2):
    increases += 1 if \
        int(input[i]) + int(input[i+1]) + int(input[i+2]) > \
        int(input[i]) + int(input[i-1]) + int(input[i+1]) else 0

print(increases)

File: 01/part2.pl
open my $file, '<', 'input.txt';

my @input;

while (<$file>) {
    chomp;
    push @input, int $_;
}
close $file;

my $increases = 0;

for (my $i = 0; $i < scalar @input - 2; $i++) {
    $increases += @input[$i] + @input[$i+1] + @input[$i+2] > @input[$i] + @input[$i-1] + @input[$i+1];
}

print $increases;
